dask/dataframe/dask_expr/_concat.py

```diff
@@ -264,12 +270,13 @@ def get_columns_or_name(e: Expr):
             result = type(self)(
                 self.join,
                 self.ignore_order,
-                self._kwargs,
                 self.axis,
                 self.ignore_unknown_divisions,
                 self.interleave_partitions,
+                self._kwargs,
                 *frames,
```

Comment:
Ordering was a bit messed up here as well. I attempted to pass the arguments here using keywords but that doesn't work. Our paramters parsing logic isn't smart enough and doesn't capture the frames properly but instead raises a cryptic error message

Reply:
Yep, this positional thing is a hack to pass a variable number of expressions….